### DISCUSSION TITLE
Wargames Movie (WOPR) Simulator written in BASIC for your IMSAI 8080esp (or other CP/M machine) + WOPR Server written in C

### DISCUSSION THREAD

**Discussion by zompiexx** [thehighnibble/thehighnibble.github.io#75] [Show and tell]
Hi All:

Just to let you know, I have published a .dsk image (for the IMSAI 8080esp) on my Github for the Wargames (WOPR) simulator that I wrote in BASIC a few years ago. Feel free to download and play with it. Everything you need is on the disk image.

I am also in the process of writing a WOPR server in C, so that it can be hosted on separate Linux machine and it will accept connections via telnet. This is already working, but limited functionality at the moment - development ongoing.

Project can be found here: https://github.com/zompiexx/wargames/

Cheers,

Andy

**Comment by rcoonjr** [comment]
This is awesome.   I was trying it on mine using thr basic version you included.  Ran out of memory.    But I noticed your YouTube video running a compiled version.  I didn't see that in the package.  I'm assuming I have to complie it myself?  Would love to use it for the vintage computer festival Midwest in September to show these imsai 8080 replicas and your program is perfect!

**Reply by zompiexx** [reply]
Hi, thanks. Yes, there is a compiled version, but it's an earlier version of the source code as there is not enough TPA memory to compile the completed code. Here is a link to the disk image for the compiled version: https://github.com/zompiexx/wargames/blob/main/BASIC/wargames-old.dsk

You might want to have a look at the C code version. This has a lot more functionality, including the voice of WOPR and Chat GPT integration (optional). Whilst it's not possible to compile the C version for the IMSAI 8080, it can be compiled on a Linux machine, such as a Raspberry PI (or other SBC) - and the Wargames servers hosted on the PI can be connected to via telnet from the IMSAI 8080esp using Kermit with the WiFi modem. There is a load more information on the C version, and linked YouTube videos on my Github page. 

https://youtu.be/18RgzBr7AFY?si=kkXdNVyS70zy0Qtb
https://youtu.be/-UFv-aeyP6E?si=gArdv5I_uVDhsd1U

Let me know if you need any help.

Cheers,

Andy

**Reply by rcoonjr** [reply]
Great .. I'm actually gonna prep a Pi 4 with Ubuntu and the programming tools.. and go ahead and compile this version.  Now connecting to it should be easy then.. I'll keep working on it.  And I really appreciate your help and work!  I'll make sure you get due credit sir!

**Reply by rcoonjr** [reply]
I got it kinds running.  But it won't let compile right..  this is what I get..![20230827_003457.jpg](https://github.com/thehighnibble/thehighnibble.github.io/assets/142746414/913ba4b6-7795-41ca-bd18-e471371e0d1a)

**Reply by rcoonjr** [reply]
Unless I need to install the gpt shell then recompile?

**Reply by zompiexx** [reply]
> Unless I need to install the gpt shell then recompile?

You'll need to install shell_gpt if you want chat gpt integration, but if not you can ignore the script error. If you don't need chat gpt, this can be commented out from the WOPR source code, then recompile. shell_gpt needs a chat gpt API key, which needs a subscription to work.

Cheers,

Andy

**Reply by zompiexx** [reply]
Hi. For the WOPR, you’ll need to install ncurses. sudo apt install libncurses5-dev. For Dialer, I’ll check the source code when I get back to my dev machine and post an update on GitHub once I’ve tweaked the code. Weirdly, it compiles fine on my dev machine, but I’ve seen the same issue on other platforms. All the other programs should work though, so you should be able to test those. If you run: ls -l in ~/wargames there should be Imsai8080, WOPR, School, Pan-Am, Bank. Hopefully that will help you a little. Cheers, Andy

P.S. getting the telnet listeners working is quite straightforward, but let me know when you get to that point and I'll provide some steps to follow.

Also, it might be worth logging a ticket on my Github page, and we can work through it there.

**Reply by zompiexx** [reply]
I have updated the install script to include all the dependencies, including libncurses5-dev, telnet and telnetd. You can modify the install.sh to comment out telnet and telnetd if you don't intend to run in client/server mode. 

I have updated the wopr.c code to fix the compiling error you were seeing. I have also commented out the Chat GPT code in the wopr.c file and install.sh, but they can be re-enabled easily if required. 

I have tested the software on a Raspberry PI4 running Raspberry OS Bullseye, and it all seems to be working fine now.

Please try it now and let me know if you have any further issues.

Cheers,

Andy

**Reply by rcoonjr** [reply]
I did get it working!  Impressive!   I have it on a pi4.  Now I can run it local there but getting the imsai replica to use it with kermit I cannot get going.  I've got them both on the same wifi (default one for the replica)  I may be able to use USB serial and possiable use the terminal on the comms disk?  unless I'm looking at that wrong?   my imsai has an actual terminal on port 1 of the replica working perfectly.  color terminal at that.

**Reply by zompiexx** [reply]
Are you able to connect the IMSAI 8080esp using Kermit and the WiFi modem to any system, e.g. a BBS? To do this, you will need the comms.dsk disk image that should have come with the IMSAI 8080esp (look in disks library). This should have kermit.com on it. Run kermit.com, then use the following commands: 

set port uc1
connect
ATDhost:port

Replace host with the hostname or IP address of the system you want to connect to, and replace port with the port number, see example below:

NOTE: You could try connecting to Simulant BBS as a test: ATDamstrad.simulant.uk:464

I have been doing quite a lot of work on the software this week, so you might want to get the latest version of code from my Github page. I have updated the install.sh script and included a load of new features, including user accounts on the WOPR and email. I have also updated the installation instructions, so you might want to check those as they detail how to use the scripts for client/server mode: https://github.com/zompiexx/wargames#readme

Cheers,

Andy
